modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java

```diff
@@ -433,9 +433,30 @@
                     }
                 }
                 else {
-                    if (log.isInfoEnabled())
+                    if (log.isInfoEnabled()) {
+                        String serverNodeId = "";
+                        String consistentId = "";
+
+                        try {
+                            Collection<ClusterNode> remoteNodes = getSpiContext().remoteNodes();
+
+                            for (ClusterNode node : remoteNodes) {
+                                if (U.toInetAddresses(node).contains(ses.remoteAddress().getAddress()) && !node.isClient()) {
```

Comment:
We could get not all possible remote nodes, if we will have a few server nodes on one host.